geomstats/geometry/product_manifold.py

```diff
@@ -668,27 +655,67 @@ def log(self, point, base_point=None, **kwargs):
             Tangent vector at the base point equal to the Riemannian logarithm
             of point at the base point.
         """
-        args = {"point": point, "base_point": base_point} | kwargs
+        args = {"point": point, "base_point": base_point}
         logs = self._iterate_over_factors("log", args)
-        return self._pool_outputs_from_function(logs)
+        return self._space.embed_to_product(logs)
 
-    def dist(self,point_a, point_b,**kwargs):
-        args = {"point_a": point_a, "point_b": point_b} | kwargs
+    def dist(self, point_a, point_b):
+        """Geodesic distance between two points.
+
+        Parameters
+        ----------
+        point_a : array-like, shape=[..., self.shape]
+            Point.
+        point_b : array-like, shape=[..., self.shape]
+            Point.
+
+        Returns
+        -------
+        dist : array-like, shape=[...,]
+            Distance.
+        """
+        args = {"point_a": point_a, "point_b": point_b}
         dists = gs.array(self._iterate_over_factors("dist", args))
-        return gs.linalg.norm(dists, ord=2, axis=0) 
-    
-    def geodesic(self, initial_point, end_point=None, initial_tangent_vec=None, **kwargs):
-        args = {"initial_point": initial_point, "end_point": end_point, "initial_tangent_vec":initial_tangent_vec} | kwargs
+        return gs.linalg.norm(dists, ord=2, axis=0)
+
+    def geodesic(self, initial_point, end_point=None, initial_tangent_vec=None):
+        """Generate parameterized function for the geodesic curve.
+
+        Geodesic curve defined by either:
+
+        - an initial point and an initial tangent vector,
+        - an initial point and an end point.
+
+        Parameters
+        ----------
+        initial_point : array-like, shape=[..., dim]
+            Point on the manifold, initial point of the geodesic.
+        end_point : array-like, shape=[..., dim], optional
+            Point on the manifold, end point of the geodesic. If None,
+            an initial tangent vector must be given.
+        initial_tangent_vec : array-like, shape=[..., dim],
+            Tangent vector at base point, the initial speed of the geodesics.
+            Optional, default: None.
+            If None, an end point must be given and a logarithm is computed.
+
+        Returns
+        -------
+        path : callable
+            Time parameterized geodesic curve. If a batch of initial
+            conditions is passed, the output array's first dimension
+            represents the different initial conditions, and the second
+            corresponds to time.
+        """
+        args = {
+            "initial_point": initial_point,
+            "end_point": end_point,
+            "initial_tangent_vec": initial_tangent_vec,
+        }
```

Comment:
Sometimes we may want to add additional arguments to geodesic. For example in SRVQuotientMetric the geodesic and distance can be computed through 2 different methods ( "iterative horizontal projection" or "dynamic programming" ) and each method needs it's own parameters to be specified. So, for example if we want to compute a geodesic (or dist) of a ProductMetric where one of the metrics is SRVQuotientMetric we may want to specifie some parameters that will need to be passed on to each geodesic. That's why kwargs was added to args

Reply:
We are trying to homogenize all signatures of methods such as `exp`, `log`, `geodesic`, and so on, to receive only the arguments that are shared among all manifolds (which means only points and tangent vectors).

Other behaviors should be controlled at init. The strategy we are using now is to create different objects implementing the different ways of solving a problem, which we are calling solvers (see e.g. #1860, where before we were passing arguments to `log` and `exp` in e.g. `DirichletMetric` and now we don't do it anymore).

Another strategy can be to use a variable at instantiation choosing the strategy (I would suggest this to be temporary though), and then use an if.

This is super important because learning methods only pass tangent vectors and points to these methods, so it is the only way to control how to compute a given quantity when using a space within them.

Reply:
Okey I see. Thanks for the explanation !